torch/csrc/autograd/function.cpp

```diff
@@ -152,7 +152,7 @@ thread_local size_t deleteFunctionRecursionDepth = 0;
 #ifdef _WIN32
 constexpr size_t kDeleteFunctionMaxRecursionDepth = 3000;
 #else
-constexpr size_t kDeleteFunctionMaxRecursionDepth = 10000;
+constexpr size_t kDeleteFunctionMaxRecursionDepth = 5000;
```

Comment:
Why did that change? Did you get any segfaults?

Reply:
This is a totally unrelated thing, I briefly changed it because we were getting errors in fbcode for it. I will revert this